src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/S3Service.kt

```diff
@@ -23,12 +24,15 @@ class S3Service(
   fun uploadMessage(uriPath: String, messageContent: String): String? {
     val s3Key = buildS3Key(
       courtCode = getCourtCode(messageContent),
-      receiptTime = LocalDateTime.now(),
+      receiptTime = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS),
       messageType = getMessageType(uriPath),
       hearingEventId = findUuid(uriPath),
     )
 
     return try {
+      if (amazonS3Client.doesObjectExist(bucketName, s3Key)) {
+        log.warn("File {} already exists in S3 bucket", s3Key)
+      }
```

Comment:
This assumes we cannot have 2 different payloads for the same hearing within a second

Reply:
FIFO work seems a more reliable approach to this

Reply:
i agree. also it does assume milliseconds don't make a difference

Reply:
suprisingly the data race gets stored in s3 as 1 json file rather than duplicates when we get rid of milliseconds